misk/src/test/kotlin/misk/client/HttpClientEnvoyTest.kt

```diff
@@ -0,0 +1,93 @@
+package misk.client
+
+import com.google.inject.Guice
+import com.google.inject.Injector
+import com.google.inject.Provides
+import com.google.inject.name.Names
+import helpers.protos.Dinosaur
+import misk.MiskServiceModule
+import misk.inject.KAbstractModule
+import misk.inject.getInstance
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.POST
+import java.io.File
+import javax.inject.Singleton
+
+/** Demonstrates ability to bind up an envoy-based http client and connect via unix sockets. */
+internal class HttpClientEnvoyTest {
+  private lateinit var socketFile: File
+  private lateinit var server: MockWebServer
+  private lateinit var clientInjector: Injector
+
+  @BeforeEach
+  fun createClientAndServer() {
+    socketFile = File("/tmp/ClientAndServer.sock")
+    server = MockWebServer()
+    server.setServerSocketFactory(UnixDomainServerSocketFactory(socketFile))
+
+    clientInjector = Guice.createInjector(
+        ClientModule(TestEnvoyClientConfiguration(socketFile, server)))
+  }
+
+  @AfterEach
+  fun stopServer() {
+    server.shutdown()
+    socketFile.delete()
+  }
+
+  @Test
+  fun useEnvoyClient() {
+    val dinoRequest = Dinosaur.Builder().name("dinoRequest").build()
+
+    server.enqueue(MockResponse())
+
+    val client: ReturnADinosaur = clientInjector.getInstance(Names.named("dinosaur"))
+    val response = client.getDinosaur(dinoRequest).execute()
+    assertThat(response.code()).isEqualTo(200)
+
+    val recordedRequest = server.takeRequest()
+    assertThat(recordedRequest.path).isEqualTo("/cooldinos")
+    assertThat(recordedRequest.getHeader("Host"))
+        .isEqualTo(String.format("%s:%s", server.hostName, server.port))
```

Comment:
I'm curious, what do these get set to? Unix sockets don't have a port?

Reply:
yeah, it's not meaningful, but a value is still required to be passed in from okhttp. Checkout https://github.com/square/misk/pull/613/files#diff-92980dcb978415a24adb016b23166945R67